default.nix

```diff
@@ -0,0 +1,21 @@
+{ config, lib, pkgs, ... }:
+
+pkgs.buildGoModule rec {
+  pname = "process-compose";
+  version = "0.10.2";
```

Comment:
Do you think there is a way to take the version value from the latest git tag?
If not I think I can add some sort of automation to the Makefile or create a git hook.

Reply:
It's usually hand-managed in cases where the source of the repository/package is remote. Since it's local, you could create a simple version.txt file with some script (outside of Nix) and reference it here with 'version = ./version.txt'

Reply:
Will this change work with both remote and local packages? Sorry if it sounds like a trivial question, I am just not familiar with the nix syntax.
Will the versions file have to be committed as well? Is there any special syntax for the contents or any "version like" string will work (e.g. v0.10.3)?

Reply:
The way it's written now it uses the local filesystems contents to build the project.
In the end if some1 wants to build it by fetching the repo via git, they can do it like this (found a random package a definition on the internet):
```nix
pet = buildGoModule rec {
  pname = "pet";
  version = "0.3.4";

  src = fetchFromGitHub {
    owner = "knqyf263";
    repo = "pet";
    rev = "v${version}";
    sha256 = "0m2fzpqxk7hrbxsgqplkg7h2p7gv6s1miymv3gvw0cz039skag0s";
  };

  vendorSha256 = "1879j77k96684wi554rkjxydrj8g3hpp0kvxz03sd8dmwr3lh83j";

  meta = with lib; {
    description = "Simple command-line snippet manager, written in Go";
    homepage = "https://github.com/knqyf263/pet";
    license = licenses.mit;
    maintainers = with maintainers; [ kalbasit ];
  };
}
```
> Will the versions file have to be committed as well?
Yes, it's best practice if you want to go this route.
> Is there any special syntax for the contents or any "version like" string will work (e.g. v0.10.3)?
Generally the version doesn't have a `v` prefix, but see for yourself:
https://github.com/NixOS/nixpkgs/search?q=version+%3D